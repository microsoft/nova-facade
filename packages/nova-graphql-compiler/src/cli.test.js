import { spawn } from "cross-spawn";
import { readFile, rm } from "fs/promises";
import { join } from "path/posix";
import { describe, it, expect } from "vitest";


describe("cli", () => {
  it("generates files", async () => {
    // Remove old generated files if they exist
    await rm(join(__dirname, "__fixtures__/src/__generated__"), {
      recursive: true,
      force: true,
    });

    // Run the compiler
    spawn.sync(
      "./src/cli.js",
      [
        "--schema",
        "./src/__fixtures__/schema.graphql",
        "--src",
        "./src/__fixtures__/src",
        "--watchman",
        "false",
      ],
      {
        stdio: "inherit",
      },
    );

    // Read the generated files
    const generatedFiles = await readFile(
      join(
        __dirname,
        "__fixtures__/src/__generated__/Example_feedbackFragment.graphql.ts",
      ),
      "utf8",
    );

    expect(generatedFiles).toMatchSnapshot();
  }, 50000);
});
