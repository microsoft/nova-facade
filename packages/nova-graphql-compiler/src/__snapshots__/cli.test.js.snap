// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cli > generates files 1`] = `
"/* tslint:disable */
/* eslint-disable */
// @ts-nocheck
;
import { FragmentRefs } from "@graphitation/apollo-react-relay-duct-tape";
export type Example_feedbackFragment = {
    readonly id: string;
    readonly message: {
        readonly text: string;
    };
    readonly doesViewerLike: boolean;
    readonly " $refType": "Example_feedbackFragment";
};
export type Example_feedbackFragment$data = Example_feedbackFragment;
export type Example_feedbackFragment$key = {
    readonly " $data"?: Example_feedbackFragment$data | undefined;
    readonly " $fragmentRefs": FragmentRefs<"Example_feedbackFragment">;
};
"
`;
